File: src/main/java/com/citasmedicas/appcitasmedicas/Service/PatientService.java
package com.citasmedicas.appcitasmedicas.Service;

import com.citasmedicas.appcitasmedicas.dto.Request.CreatePatientRequest;
import com.citasmedicas.appcitasmedicas.dto.Request.UpdatePatientRequest;
import com.citasmedicas.appcitasmedicas.dto.Response.PatientResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PatientService {
    
    PatientResponse create(CreatePatientRequest req);
    
    PatientResponse update(Long id, UpdatePatientRequest req);
    
    PatientResponse findById(Long id);
    
    Page<PatientResponse> findAll(Pageable pageable);
    
    void delete(Long id);
}

File: frontend/src/services/patientService.js
import api from './api';

export const patientService = {
  getAll: (page = 0, size = 100) =>
    api.get(`/patients?page=${page}&size=${size}`),

  getById: (id) =>
    api.get(`/patients/${id}`),

  create: (data) =>
    api.post('/patients', data),

  update: (id, data) =>
    api.patch(`/patients/${id}`, data),

  delete: (id) =>
    api.delete(`/patients/${id}`),
};